client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchRoundDataEntry.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import { Callout } from '@blueprintjs/core'
+
+import BatchDetails from './BatchDetails'
+import useContestsJurisdictionAdmin from '../useContestsJurisdictionAdmin'
+import { Confirm, useConfirm } from '../../Atoms/Confirm'
+import {
+  Detail,
+  List,
+  ListAndDetail,
+  ListItem,
+} from '../../Atoms/ListAndDetail'
+import {
+  IBatch,
+  IBatchResultTallySheet,
+  useBatches,
+  useRecordBatchResults,
+} from '../useBatchResults'
+import { useDebounce } from '../../../utils/debounce'
+
+const Container = styled('div')`
+  display: flex;
+  flex-direction: column;
+  gap: 12px;
+  width: 100%;
+`
+
+interface IProps {
+  electionId: string
+  jurisdictionId: string
+  roundId: string
+}
+
+const BatchRoundDataEntry: React.FC<IProps> = ({
+  electionId,
+  jurisdictionId,
+  roundId,
+}) => {
+  const batchesQuery = useBatches(electionId, jurisdictionId, roundId)
+  const contestsQuery = useContestsJurisdictionAdmin(electionId, jurisdictionId)
+  const recordBatchResults = useRecordBatchResults(
+    electionId,
+    jurisdictionId,
+    roundId
+  )
+  const { confirm, confirmProps } = useConfirm()
+
+  const [searchQuery, setSearchQuery] = useState('')
+  const [debouncedSearchQuery] = useDebounce(searchQuery)
+  const [selectedBatchId, setSelectedBatchId] = useState<IBatch['id']>()
+  const [areChangesUnsaved, setAreChangesUnsaved] = useState(false)
+
+  const batches = batchesQuery.isSuccess ? batchesQuery.data.batches : []
+  const filteredBatches = batches.filter(batch =>
+    batch.name.toLowerCase().includes(debouncedSearchQuery.toLowerCase())
+  )
+  const selectedBatch = batches.find(batch => batch.id === selectedBatchId)
+
+  // Auto-select first batch on initial load
+  useEffect(() => {
+    if (!selectedBatchId && filteredBatches.length > 0) {
+      setSelectedBatchId(filteredBatches[0].id)
+    }
+  }, [filteredBatches, selectedBatchId, setSelectedBatchId])
+
+  // Auto-select first search match
+  useEffect(() => {
+    if (
+      debouncedSearchQuery &&
+      filteredBatches.length > 0 &&
+      !areChangesUnsaved
+    ) {
+      setSelectedBatchId(filteredBatches[0].id)
+    }
+  }, [
+    debouncedSearchQuery,
+    filteredBatches,
+    areChangesUnsaved,
+    setSelectedBatchId,
+  ])
```

Comment:
I played around with removing this effect, but the resulting code felt more complex and prone to bugs than the effect (which I think is fairly easy to reason about for an effect), so I kept things as is.

Trying to use the search query onChange for everything is a bit tricky in this case because we have one data item that needs to be updated immediately (the search query) and another that needs to be debounced (the list of filtered batches / selected batch ID). Also, because we need the list of filtered batches when we render, not just the selected batch ID, I was having to either duplicate the list of filtered batches outside the search query onChange and inside it, or put it in state, both non-ideal.